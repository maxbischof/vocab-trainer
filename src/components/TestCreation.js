import React, { useState } from 'react'
import styled from 'styled-components'
import TestLink from './TestCreated'
import Button from './ui/Button'
import { createTest } from '../lib/firebase'
import List from './wordpair/List'
import Form from './wordpair/Form'

export default function TestCreation({ db }) {
  const [wordPairs, setWordPairs] = useState([])
  const [testURL, setTestURL] = useState()

  function deleteWord(index) {
    const newWords = [...wordPairs]
    newWords.splice(index, 1)
    setWordPairs(newWords)
  }

  return (
    <>
      {testURL ? (
        <TestLink testURL={testURL} />
      ) : (
        <StyledTestCreator>
          <h1>Test erstellen</h1>
          <Form wordPairs={wordPairs} setWordPairs={setWordPairs} />

          {wordPairs.length > 0 && (
            <TestPreview>
              <Button onClick={() => createTest(wordPairs, db, setTestURL)}>
                Test erstellen
              </Button>
              <List words={wordPairs} deleteWord={deleteWord} />
            </TestPreview>
          )}
        </StyledTestCreator>
      )}
    </>
  )
}

const TestPreview = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid var(--text);
  border-radius: 10px;
  padding: 10px;
  margin: 40px;
`

const StyledTestCreator = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`
