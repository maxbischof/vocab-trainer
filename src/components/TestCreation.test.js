import React from 'react'
import TestCreation from './TestCreation'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

it('renders correctly', () => {
  const tree = renderer.create(<TestCreation />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('add wordpairs to the list', () => {
  render(<TestCreation />)
  userEvent.type(screen.getByPlaceholderText('Fremdsprache'), 'the car')
  userEvent.type(screen.getByPlaceholderText('Deutsch'), 'das Auto')
  userEvent.click(screen.getByText('Wortpaar hinzufügen'))

  expect(screen.getByText(/the car/)).toBeInTheDocument()

  expect(screen.getByText(/das Auto/)).toBeInTheDocument()
})

it('removes wordpairs from the list', () => {
  render(<TestCreation />)
  userEvent.type(screen.getByPlaceholderText('Fremdsprache'), 'the car')
  userEvent.type(screen.getByPlaceholderText('Deutsch'), 'das Auto')
  userEvent.click(screen.getByText('Wortpaar hinzufügen'))
  userEvent.click(screen.getByRole('listitem').querySelector('svg'))

  expect(screen.queryByText(/the car/)).toBeNull()
  expect(screen.queryByText(/das Auto/)).toBeNull()
})
