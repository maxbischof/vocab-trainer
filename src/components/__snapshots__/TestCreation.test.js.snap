// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-dQppl cSyhzl"
>
  <h1>
    Test erstellen
  </h1>
  <form
    className="sc-iqHYGH bQkWaJ"
    onSubmit={[Function]}
  >
    <input
      className="sc-iBPRYJ jxdMbx"
      name="foreign"
      onChange={[Function]}
      placeholder="Foreign"
      type="text"
      value=""
    />
    <input
      className="sc-iBPRYJ jxdMbx"
      name="native"
      onChange={[Function]}
      placeholder="Native"
      type="text"
      value=""
    />
    <div
      className="sc-pFZIQ mxfyr"
    >
      <div
        className="sc-jrAGrp dwsgCp"
      >
        <div
          className="sc-fubCfw fYNuti"
          color="var(--masculine)"
          fontSize={21.5}
          onClick={[Function]}
        >
          m
        </div>
        <div
          className="sc-fubCfw gDpyfe"
          color="var(--feminine)"
          fontSize={21.5}
          onClick={[Function]}
        >
          f
        </div>
        <div
          className="sc-fubCfw NlhyU"
          color="var(--neuter)"
          fontSize={21.5}
          onClick={[Function]}
        >
          n
        </div>
      </div>
    </div>
    <div
      className="sc-kEjbxe hLcugI"
    >
      <div
        className="sc-fubCfw cZCdYp"
        color="var(--noun)"
        fontSize={31.5}
        onClick={[Function]}
      >
        N
      </div>
      <div
        className="sc-fubCfw gCDOfH"
        color="var(--adjective)"
        fontSize={31.5}
        onClick={[Function]}
      >
        A
      </div>
      <div
        className="sc-fubCfw bNrrpL"
        color="var(--verb)"
        fontSize={31.5}
        onClick={[Function]}
      >
        V
      </div>
    </div>
    <button
      className="sc-gsTCUz hwVgqy"
      type="submit"
    >
      Add word
    </button>
  </form>
</div>
`;
